src/rudder-by-example/modules/ROOT/pages/files/advanced-file-templating.adoc

```diff
@@ -116,4 +114,103 @@ http {
         server server6.rudder.local weight=1;
     }
 }
-----
\ No newline at end of file
+----
+
+== Advanced example: sshd_config templating
+
+This example ensures the content of a bastillion host's `sshd_config`, configurable via either variables shared via group membership or private node properties.
+
+=== Source data
+
+We can either define the following node property in our Rudder server for any secured system, or use a `Variables (any)` directive to assing this to multiple systems in a group (like systems requiring extra security due being exposed to the internet).
+
+Value of either *Node property* named `ssh_access` or a *Variable (any)* Directive named `bastillion.ssh_access`:
+
+[source,json]
+----
+{ 
+  "fromip" : [ "10.111.20.1", "10.111.20.2" ],
+  "user"   : [ "joe", "jack" ],
+  "group"  : [ "admins" ]
+}
+----
+
+=== Configuration policy
+
+The configuration will consist of a shared template downloaded from `shared-files` of the rudder server and a method to provide the input data for the template. We will show two approaches on that.
+
+==== Example template with a *Node property* holding the data
+
+[source,mustache]
+----
+Protocol 2
+PasswordAuthentication no
+PubkeyAuthentication yes
+AuthenticationMethods publickey
+DenyUsers *
+
+Match Address {{#vars.node.properties.ssh_access.fromip}}{{{.}}},{{/vars.node.properties.ssh_access.fromip}} User {{#vars.node.properties.ssh_access.user}}{{{.}}},{{/vars.node.properties.ssh_access.user}}
+  DenyUsers !*
+
+Match Address {{#vars.node.properties.ssh_access.fromip}}{{{.}}},{{/vars.node.properties.ssh_access.fromip}} Group {{#vars.node.properties.ssh_access.group}}{{{.}}},{{/vars.node.properties.ssh_access.group}}
+  DenyUsers !*
+----
+
+The iterator of the arrays uses the `{{{.}}}` to reference the value if the currently iterated item, which is the value of the array item.
+
+==== Example template with a *Variables (any)* Directive holding the data.
+
+You can use a dedicated Rule to assign the Directive to a group of nodes, with this approach you can have different Directives allowing different access but reuse the same template and NCF Directive with data provided by different Rules.
+
+[source,mustache]
+----
+Protocol 2
+PasswordAuthentication no
+PubkeyAuthentication yes
+
+DenyUsers *
+
+Match Address {{#vars.bastillion.ssh_access.fromip}}{{{.}}},{{/vars.bastillion.ssh_access.fromip}} User {{#vars.bastillion.ssh_access.user}}{{{.}}},{{/vars.bastillion.ssh_access.user}}
+  DenyUsers !*
+
+Match Address {{#vars.bastillion.ssh_access.fromip}}{{{.}}},{{/vars.bastillion.ssh_access.fromip}} Group {{#vars.bastillion.ssh_access.group}}{{{.}}},{{/vars.bastillion.ssh_access.group}}
+  DenyUsers !*
+----
+
+==== NCF Techniqe to deploy the template
```

Comment:
Technique

Reply:
We try to remove the usage of "NCF" in Rudder interface and documentation as it does not mean much to users who don't know Rudder's internals. We now usually simply use "technique", and "pre-built technique" to distinguish those coming from the technique editor from those coming from `rudder-techniques`.